### ISSUE TITLE
Question: TargetPort in OSM TrafficSplit

### ISSUE THREAD

**Issue by ruokun-niu** [openservicemesh/osm#4878]
Hi OSM repo,

Currently I am working on a project that utilizes the traffic-splitting feature from Open Service Mesh, and I have a question regarding the TargetPort property of the traffic split root service.

According to [SMI's documentation](https://github.com/servicemeshinterface/smi-spec/blob/main/apis/traffic-split/v1alpha2/traffic-split.md#ports), in order to achieve traffic-splitting, we need to ensure that the TargetPort of the root traffic split service matches with the ports of the backend services (traffic will be directed to these services). The TargetPort of those backend services will need to match the ContainerPort of the pods.

However, as I was trying out the [Canary Rollout demo](https://release-v1-0.docs.openservicemesh.io/docs/demos/canary_rollout/) from the official OSM site, I noticed that the TargetPort of the root traffic split service needs to match with the ContainerPort of the pods. This demo uses `curl` to check if traffic-splitting is achieved. It is possible that the having the TargetPort of the root service to be matching with the pods' ContainerPort is a requirement of `curl`, not OSM TrafficSplit.

My main question is as follows: Do we need the TargetPort of the root TrafficSplit service to match with the ContainerPort of the pods? I believe that we do not need this match, but I would love to have someone else confirm this.

Thanks in advance.

**Comment by ruokun-niu** [comment]
Hi @shashankram , thank you for the prompt response. I am creating a root service with no pods backing it. So in this case, I do not need to configure the TargetPort so that it is matching the pods' ContainerPort correct?

**Comment by ruokun-niu** [comment]
Yeah that supports my understanding as well. I was also able to implement TrafficSplit without configuring the TargetPort. I guess configuring the TargetPort is only needed by cURL, not OSM

**Comment by shashankram** [comment]
Thanks for confirming @ruokun-niu. Is there anything else pending on this issue or can it be marked as resolved?

**Comment by ruokun-niu** [comment]
Hi @shashankram, thank you for your help. I'll close this issue now

**Comment by shashankram** [comment]
On investigating this issue further, I do see an assumption in the code where we use the TargetPort value on the leaf service and set it as the value of the TargetPort on the root service in internal objects while building the route configuration objects within the control plane. 

I'll investigate what it takes to relax the TargetPort requirement to be the same as the TargetPort on the leaf service.
